### DISCUSSION TITLE
[solved] Automerge does not work

### DISCUSSION THREAD

**Discussion by jokay** [renovatebot/renovate#9193] [Request Help]
When I use this configuration the automerge does not work, the user which executes renovate (the pipeline) has all necessary permissions.

```json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "config:base",
    "docker:enableMajor"
  ],
  "labels": [
    "dependencies"
  ],
  "prHourlyLimit": 0,
  "prConcurrentLimit": 0,
  "automerge": true,
  "requiredStatusChecks": null
}
```

tried this as well without success

```json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "config:base",
    "docker:enableMajor"
  ],
  "packageRules": [
    {
      "matchUpdateTypes": [
        "major",
        "minor",
        "patch",
        "pin",
        "digest"
      ],
      "automerge": true,
      "requiredStatusChecks": null
    }
  ],
  "labels": [
    "dependencies"
  ],
  "prHourlyLimit": 0,
  "prConcurrentLimit": 0
}
```

any help would be much appreciated.

**Comment by rarkins** [comment]
Describe what you mean by "does not work"?

**Reply by jokay** [reply]
Yes, it creates a merge request and does not merge it automatically.

**Reply by jokay** [reply]
![sample-bot](https://user-images.githubusercontent.com/18613935/111599717-e03d6780-87d0-11eb-80f3-baacb0284f5d.png)

**Reply by viceice** [reply]
Automerge will need atlest two runs to automerge

**Reply by jokay** [reply]
Even with the option `"requiredStatusChecks": null` ?

**Reply by viceice** [reply]
yes.

**Reply by viceice** [reply]
BTW: It's a very bad idea to combine `"requiredStatusChecks": null` with `automerge=true`. You can get a broken baseBranch!

**Reply by jokay** [reply]
Ok, on a subsequent run it just closes the merge request.

```sh
INFO: PR automerged (repository=renovate/renovate, branch=renovate/docker-renovate-renovate-24.x)
       "pr": 14,
       "prTitle": "Update renovate/renovate Docker tag to v24.89.1"
```

```sh
The changes were not merged into main

This project requires squashing commits when merge requests are accepted. Try again.
```

Is there any way to support **squashing commits** ?

**Reply by jokay** [reply]
> BTW: It's a very bad idea to combine `"requiredStatusChecks": null` with `automerge=true`. You can get a broken baseBranch!

Yes, if this needs to be run twice anyway, I can keep the checks. This automerge is only enabled for the renovate repo :)

**Reply by jokay** [reply]
Ok, there exists already issue [#7859](https://github.com/renovatebot/renovate/issues/7859) for it.
